File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/Binary.java
// Binary.java

public abstract class Binary extends Operator
{
	public Binary(Expression aOperand1, Expression aOperand2)
	{
		setOperand1(aOperand1);
		setOperand2(aOperand2);
	}

	public abstract void accept(ExpressionVisitor aExpressionVisitor);

	private void setOperand1(Expression aOperand1)
	{
		myOperand1 = aOperand1;
	}

	public Expression getOperand1()
	{
		return myOperand1;
	}

	private void setOperand2(Expression aOperand2)
	{
		myOperand2 = aOperand2;
	}

	public Expression getOperand2()
	{
		return myOperand2;
	}

	private Expression myOperand1;
	private Expression myOperand2;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/binary.cpp
// binary.cpp

#include "binary.h"
class expression;

binary::binary(expression * a_operand1, expression * a_operand2)
{
	set_operand1(a_operand1);
	set_operand2(a_operand2);
}

void binary::set_operand1(expression * a_operand1)
{
	my_operand1 = a_operand1;
}

expression * binary::get_operand1() const
{
	return my_operand1;
}

void binary::set_operand2(expression * a_operand2)
{
	my_operand2 = a_operand2;
}

expression * binary::get_operand2() const
{
	return my_operand2;
}
